dot/sync/disjoint_block_set.go

```diff
@@ -80,13 +103,42 @@ type disjointBlockSet struct {
 
 	// map of parent hash -> child hashes
 	parentToChildren map[common.Hash]map[common.Hash]struct{}
+
+	timeNow func() time.Time
 }
 
 func newDisjointBlockSet(limit int) *disjointBlockSet {
 	return &disjointBlockSet{
 		blocks:           make(map[common.Hash]*pendingBlock),
 		parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
 		limit:            limit,
+		timeNow:          time.Now,
+	}
+}
+
+func (s *disjointBlockSet) start(ctx context.Context, done chan<- struct{}) {
+	defer close(done)
+	ticker := time.NewTicker(clearBlocksInterval)
+
+	for {
+		select {
+		case <-ticker.C:
+			s.clearBlocks()
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		}
+	}
+}
+
+func (s *disjointBlockSet) clearBlocks() {
+	s.Lock()
+	defer s.Unlock()
+
+	for _, block := range s.blocks {
+		if time.Since(block.clearAt) > 0 {
```

Comment:
Use the `s.timeNow()` for testability. From the top of my head this should work

```suggestion
		if s.timeNow().Add(-block.clearAt) > 0 {
```